### PR TITLE
Add tool to random generate feature importances

### PR DESCRIPTION
**Pull request by yuhong-sigopt**

### CODE REVIEW

File: tools/generate_feature_importances.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import string
+import numpy as np
+import sys
+
+def rand_str(n, chars=string.ascii_letters + string.digits):
+  chars = np.array(list(chars))
+  idx = np.random.choice(len(chars), n)
+  return ''.join(chars[idx])
+
+def generate_feature_importances(num_feature=50, max_feature_len=100, score_type='exp'):
```
Comment (harvey-intel) [inline tools/generate_feature_importances.py:11]: `max_feature_len` is unused?

**Review by harvey-intel — COMMENTED** [review]
Is the `tool/` directory just for us?

File: tools/generate_feature_importances.py (hunk 1 above)
Comment (yuhong-sigopt) [reply tools/generate_feature_importances.py:11]: Fixed.

**Comment by yuhong-sigopt** [comment]
> Is the `tool/` directory just for us?

No. I think it is used for all kinds of tools.

File: sigopt/__init__.py [hunk 2]
```diff
@@ -47,3 +48,12 @@ def set_project(project):
...
   return _global_factory.set_project(project)
+
+def switch_run_context(run_id):
+  if get_run_id() == run_id:
+    return
+
+  context = RunContext(
+    connection = get_connection(),
+    run = get_run(run_id))
+  _global_run_context._run_context = context
```
Comment (taylorjacklespriggs) [inline sigopt/__init__.py:59]: I don't think this needs to be a global function for this PR. Please initialize the run context manually in `tools/generate_feature_importances.py`.

File: sigopt/__init__.py [hunk 3]
```diff
@@ -19,6 +19,7 @@
 log_metric = _global_run_context.log_metric
 log_metrics = _global_run_context.log_metrics
 log_model = _global_run_context.log_model
+log_sys_metadata = _global_run_context.log_sys_metadata
```
Comment (taylorjacklespriggs) [inline sigopt/__init__.py:22]: please remove

Comment (yuhong-sigopt) [reply sigopt/__init__.py]: Done.

File: sigopt/__init__.py (hunk 2 above)
Comment (yuhong-sigopt) [reply sigopt/__init__.py]: Done.

File: tools/generate_feature_importances.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  import argparse
+  parser = argparse.ArgumentParser()
+  parser.add_argument('--run', default=19, help='run id')
+  parser.add_argument('--score_type', default='exp', help='score type')
+  parser.add_argument('--num_feature', default=50, help='number of defauts')
+  parser.add_argument('--max_feature_length', default=100, help='max feature name length')
+  args = parser.parse_args()
+  switch_run_context(args.run)
+  fp = generate_feature_importances(args.num_feature, args.max_feature_length, args.score_type)
+  sigopt._global_run_context.log_sys_metadata('feature_importances', fp)
```
Comment (taylorjacklespriggs) [inline tools/generate_feature_importances.py:40]: I think this does the same thing and doesn't involve mucking around with the internals of the global run context

```python
context = RunContext(
    connection = get_connection(),
    run = sigopt.get_run(run_id))
fp = generate_feature_importances(args.num_feature, args.max_feature_length, args.score_type)
context.log_sys_metadata('feature_importances', fp)
```

Comment (yuhong-sigopt) [reply tools/generate_feature_importances.py:40]: I define `switch_run_context` to change context, so we can use other APsI sigopt, that is why I first put it in __init__.py, which may be useful.

Comment (yuhong-sigopt) [reply tools/generate_feature_importances.py]: Fixed.

**Review by taylorjacklespriggs — APPROVED** [review]

**Pull request merged**